aerofiles/seeyou/reader.py

```diff
@@ -193,3 +193,90 @@ def decode_description(self, description):
             return None
 
         return description
+
+    def decode_taskname(self, fields):
+        if not fields:
+            raise ParserError('Name field must not be empty')
+
+        return fields[0]
+
+    def decode_tasklist(self, fields, waypoints):
```

Comment:
> what is the difference between the two and what is the function of the converter?

The `Reader` is reading as close to the original file format as possible, the `Converter` is converting that to a more canonical data structure that can be used across all readers/writers in aerofiles.

This is somewhat similar to the Flarm/SeeYou task converter thing. You read the SeeYou task from file in a SeeYou specific structure, convert it to a generic structure and then write that generic structure to a Flarm config file (possibly through another Generic -> Flarm converter).

> should both tests be performed on the new reader function?

if I remember correctly the `Converter` is using the `Reader` internally. ideally the tests shouldn't change at all because we want to stay compatible with the current API, we only want to add new tests for the new API.